Add support for libuast 1.3

abeaumont opened the pull request.

juanjux commented on bblfsh/pyuast.c:
```diff
@@ -14,77 +14,231 @@ static const char *ReadStr(const void *data, const char *prop)
+  if (!offset) {
+    return 0;
+  }
+  return PyLong_AsUnsignedLong(offset);
```
I think a cast would be more explicit here since `PyLong_AsUnsignedLong` returns an `unsigned long`:

`return (uint32_t)PyLong_AsUnsignedLong(offset);`

Same for the other `uint32_t` functions.

abeaumont replied:
I didn't do it since I thought function signature was explicit enough in this case. I can add it though.

juanjux commented on bblfsh/pyuast.c:
```diff
@@ -14,77 +14,231 @@ static const char *ReadStr(const void *data, const char *prop)
 
-static void *ChildAt(const void *data, int index)
-{
+static void *ChildAt(const void *data, int index) {
```
I know it was that way before, but some interface functions use `*node` as parameter and others use `*data`.

abeaumont replied:
Well, since I started with style fixes I should also fix that =)

abeaumont commented:
Have simplified the code, made it more consistent, and added review feedback.

juanjux commented:
I like it, I'll adapt those remade helper functions (`Attributte, String, Size`, etc) for the Scala client too. 

Can it be merged? (the CI is working).

abeaumont commented:
Not yet, current testsuite doesn't cover much about uast filter, I want to improve that before merging.

abeaumont commented:
@juanjux added the tests and now there're random segfaults, even if all the tests pass individually. I think current bindings have some kind of memory handling problem.

abeaumont commented:
Found and fixed the reference counter problem, added some more tests, and now everything seems to work properly. I'd say this is ready to be merged :)

juanjux commented:
@alfredo, before your last message I was about to tell you that the Scala part is working well now,  but I want to torture test it a little more and maybe do a memory profiling with Valgrind to see if there is any leak (at least with the Python client that should use refcounting, the JVM is less deterministic and thus harder to evaluate for leaks). Anyway I'm tagging so people can start using the new filters.
